static/src/javascripts/projects/common/modules/ui/subscription-banner.js

```diff
@@ -52,7 +60,29 @@ const currentRegion: ReaderRevenueRegion = getReaderRevenueRegion(
     geolocationGetSync()
 );
 const hideBannerInTheseRegions: ReaderRevenueRegion[] = ['australia'];
-const subscriptionUrl = `${subscriptionHostname}/subscribe/digital?INTCMP=gdnwb_copts_banner_subscribe_SubscriptionBanner&acquisitionData=%7B%22source%22%3A%22GUARDIAN_WEB%22%2C%22campaignCode%22%3A%22subscriptions_banner%22%2C%22componentType%22%3A%22${COMPONENT_TYPE}%22%2C%22componentId%22%3A%22${CLICK_EVENT_CTA}%22%7D`;
+
+const abTest = isInABTestSynchronous(subscriptionsBannerNewYearCopyTest)
+    ? {
+          abTest: {
+              name: subscriptionsBannerNewYearCopyTest.id,
+              variant: isInVariantSynchronous(
+                  subscriptionsBannerNewYearCopyTest,
+                  'control'
+              )
+                  ? 'control'
+                  : 'variant',
+          },
+      }
+    : {};
+
+const subscriptionUrl = addTrackingCodesToUrl({
+    base: `${subscriptionHostname}/subscribe/digital`,
+    componentType: COMPONENT_TYPE,
+    componentId: OPHAN_EVENT_ID,
+    campaignCode: CAMPAIGN_CODE,
+    ...abTest,
+});
+
 const signInUrl = `${signinHostname}/signin?utm_source=gdnwb&utm_medium=banner&utm_campaign=SubsBanner_Exisiting&CMP_TU=mrtn&CMP_BUNIT=subs`;
```

Comment:
Could construct this url in the same way as above?

Reply:
Unfortunately the tracking codes helper is very specific to acquisition parameters and so it wouldn't help much with the sign-in URL. There's still a better way, but I'll leave that for another PR/time.